src/frontend/src/components/nav/MainMenu.tsx

```diff
@@ -4,46 +4,60 @@ import {
   IconChevronDown,
   IconHeart,
   IconLogout,
+  IconPlugConnected,
   IconSettings,
-  IconUserCircle
+  IconUserCircle,
+  IconUserCog
 } from '@tabler/icons-react';
+import { useContext } from 'react';
 import { Link } from 'react-router-dom';
 
+import { UserContext } from '../../contexts/UserContext';
 import { doClassicLogout } from '../../functions/auth';
 import { InvenTreeStyle } from '../../globalStyle';
-import { useApiState } from '../../states/ApiState';
-import { PlaceholderPill } from '../items/Placeholder';
 
 export function MainMenu() {
   const { classes, theme } = InvenTreeStyle();
-  const [username] = useApiState((state) => [state.user?.name]);
+
+  const user = useContext(UserContext);
```

Comment:
Question: Why this switch? Contexts are less efficient with rerenders than using a zustand state. See https://medium.com/@viraj.vimu/react-context-api-vs-zustand-state-manager-98ca9ac76904

Reply:
Thanks, I'll look into that. I was trying to work out how your implementation worked, if that is more efficient then I am all for it.